group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1683,7 +1685,7 @@ public void scheduleUnloadOperation(
             if (context != null) {
                 try {
                     context.lock.lock();
-                    if (context.epoch < partitionEpoch) {
+                    if (!partitionEpoch.isPresent() || context.epoch < partitionEpoch.getAsInt()) {
```

Comment:
did we want to have special logging for the None/empty case? I see we log partitionEpoch in a few places in this method.

Reply:
i thought that logging the empty optional in those places would be enough. would you agree?

Reply:
Ok -- I guess it is similar to logging -1, I guess folks can assume this is a deleted topic and not an error.